16/danielmichaels/weather.py

```diff
@@ -0,0 +1,72 @@
+import datetime
+import sys
+
+from api_secrets import WEATHER_TOKEN
+
+import requests
+import requests_cache
+
+from requests.exceptions import RequestException
+
+
+class Weather:
+    """A simple class that pulls in weather data from the openweathermap api and formats the response.
+
+    """
+
+    def __init__(self, city="london", country="gb"):
+        self.city = city
+        self.country = country
+        self.url = (
+            "http://api.openweathermap.org/data/2.5/weather?q={city},"
+            "{country}&units=metric&appid={TOKEN}".format(
+                city=self.city, country=self.country, TOKEN=WEATHER_TOKEN
+            )
+        )
+        self.get_data
+
+    @property
+    def get_data(self):
+        """Get the weather data in JSON format. Caches the response for
+        600 seconds in a sqlite database.
+
+        """
+        requests_cache.install_cache(expire_after=600)
+        try:
+            response = requests.get(self.url)
+            response.raise_for_status()
+            data = response.json()
+            return self.format(data)
+        except requests.exceptions.RequestException as e:
+            # for brevity don't try/except all possible cases
+            print(f"[!] An Error Occurred [!]\n\n{e}")
+            sys.exit(1)
+
+    @staticmethod
```

Comment:
interest, could be plain function outside of the class, it does not deal with class variables

Reply:
I agree. My reasoning - I think, I wrote this a while back - was to keep the method within the class because they flow together. I don't know if its an anti-pattern but I still will often keep a staticmethod in a class, if it won't be used elsewhere (and if in future it does, I'll then refactor it out) and if it fits within the overall picture of the class.

What's your opinion, is it an anti-pattern that I should avoid?